cmd/cluster/dynatrace/vault.go

```diff
@@ -0,0 +1,58 @@
+package dynatrace
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+)
+
+const (
+	vaultPath string = "osd-sre/dynatrace/sd-sre-platform-oauth-client-grail"
+	vaultAddr string = "https://vault.devshift.net"
+)
+
+type VaultResponse struct {
+	Data struct {
+		Data map[string]interface{} `json:"data"`
+	} `json:"data"`
+}
+
+func getSecretFromVault(clientID string) (secret string, error error) {
+	err := os.Setenv("VAULT_ADDR", vaultAddr)
+	if err != nil {
+		fmt.Printf("Error setting environment variable: %v\n", err)
+		return "", err
+	}
+	vaultCmd := exec.Command("vault", "login", "-method=oidc", "-no-print")
+	vaultCmd.Stdout = nil
+	vaultCmd.Stderr = nil
+	if err = vaultCmd.Run(); err != nil {
+		fmt.Println("Error running 'vault login':", err)
+		return "", nil
```

Comment:
**nitpick:** here and following, there's still errors that are just printed but not returned. Is there a reason we're not returning them to let them be returned in the main logic?